module/addRemove.js

```diff
@@ -0,0 +1,103 @@
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+let editing = -1;
+const toDoInput = document.querySelector('#new-list');
+const todoListElement = document.querySelector('.to-do-list');
+
+const saveToDos = () => {
+  const todovalue = toDoInput.value;
+  const isDuplicate = todos.some((todo) => todo.value === todovalue);
+  if (todovalue === '') {
+    return;
+  } else if (isDuplicate) {
+    return;
+  } else {
+    if (editing >= 0) {
+      todos = todos.map((todo, index) => {
+        return {
+          value: index === editing ? todovalue : todo.value,
+          checked: todo.checked,
+          color: todo.color,
+          index: index,
+        };
+      });
+      editing = -1;
+    } else {
+      todos.push({
+        value: todovalue,
+        checked: false,
+        color: '#' + Math.floor(Math.random() * 16777215).toString(16),
+        index: todovalue.id,
+      });
+      toDoInput.value = '';
+    }
+  }
+};
+const displayTodos = () => {
+  if (todos.length === 0) {
+    todoListElement.innerHTML =
+      '<center>Nothing to do yet! please add to do list </center>';
+    return;
+  }
+  todoListElement.innerHTML = '';
+  todos.forEach((todo, index) => {
+    todoListElement.innerHTML += `
+      <div class="todo" id="${index}">
+      <input type="checkbox" class="chech-box" data-action="check" />
+      <p class="list-element" data-action="check" >${todo.value}</p>
+      <i class="fa-regular fa-pen-to-square" data-action="edit" ></i>
+      <i class="fa-sharp fa-solid fa-trash" data-action="delete"></i>
+    </div>
+        `;
+  });
+};
+
+todoListElement.addEventListener('click', (event) => {
+  const target = event.target;
+  const parentElement = target.parentNode;
+
+  if (parentElement.className !== 'to-do-list') {
+    const todo = parentElement;
+    const todoId = Number(todo.id);
+    const action = target.dataset.action;
+    action === 'check' && checkTodo(todoId);
+    action === 'edit' && editTodo(todoId);
+    action === 'delete' && deleteTodo(todoId);
+  }
+});
+
+const checkTodo = (todoId) => {
+  todos = todos.map((todo, index) => {
+    if (index === todoId) {
+      return {
+        value: todo.value,
+        color: todo.color,
+        checked: !todo.checked,
+        index: index,
+      };
+    } else {
+      return {
+        value: todo.value,
+        color: todo.color,
+        checked: todo.checked,
+        index: index,
+      };
+    }
+  });
+  displayTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
```

Comment:
- [ ] Please make sure that the index gets stored in the local storage, as per [this](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=By%20default%20new%20tasks%20should%20have%20the%20property%20completed%20set%20to%20false%20and%20the%20property%20index) requirement.
![image](https://user-images.githubusercontent.com/62290545/214604051-aafa2f52-5fbc-4bf7-ba35-65f6465d3c7f.png)

Reply:
- [ ] [REQUIRED] Well done 👍 I can now see the index in the Local storage, however, there is still one core requirement you are missing out here. The property index should be set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal 5). The first index starting from 0 should start from 1 not 0.
- You can fix this by adding 1 to your index.
### Here is it from the project requirements:
<img width="911" alt="Screenshot 2023-01-27 at 13 38 39" src="https://user-images.githubusercontent.com/58692590/215089258-f9471039-5c21-400b-af0e-fbe5d26d11ac.png">

### Here is what we currently have in your Local storage
<img width="773" alt="Screenshot 2023-01-27 at 13 27 10" src="https://user-images.githubusercontent.com/58692590/215089407-13b2ecf2-9066-4dae-8a0a-7e56d45cb0af.png">